crates/typst-syntax/src/set.rs

```diff
@@ -39,54 +39,55 @@ const fn bit(kind: SyntaxKind) -> u128 {
     1 << (kind as usize)
 }
 
+/// Generate an inline const syntax set of the given kinds.
+///
+/// Example: `if syntax_set!(Plus, Minus).contains(SyntaxKind::Plus) { }`
+/// `if {const SET:SyntaxSet = SyntaxSet::new()
+///     .add(SyntaxKind::Plus)
+///     .add(SyntaxKind::Minus);
+///     SET
+/// }.contains(SyntaxKind::Plus) { }`
+macro_rules! syntax_set {
+    ( $($kind:ident),* $(,)? ) => {
+        { // 'SET' never escapes this inner block
+            const SET: crate::set::SyntaxSet = crate::set::SyntaxSet::new()
+                $(.add(crate::SyntaxKind:: $kind))*;
+            SET
+        }
+    }
+}
+
+// Export so other modules can import as: `use set::syntax_set`
+pub(crate) use syntax_set;
+
+/// Just build a syntax set of the given kinds. Only used in this file.
+///
+/// Example: `const SET: SyntaxSet = make_set!(Plus, Minus);`
+/// Becomes:
+/// `SyntaxSet::new().add(SyntaxKind::Plus).add(SyntaxKind::Minus)`
+macro_rules! make_set {
```

Comment:
Why are there two macros? They seem to be doing the same thing, one with the extra const. We can use the const one to ensure that it's really constant, but why do we need this one?

Reply:
Just that it felt redundant to — after expansion — create the in-module macros as `const SOME_SET: SyntaxSet = { const SET: SyntaxSet = ...; SET }`. Also only `syntax_set!` is exported, so this one is local to only this file.
I'm fine with just using one.

Reply:
I think the redundancy after expansion is unproblematic. If you take a look at the output of `derive(PartialEq)`: It's much worse ...

Reply:
The cleanest solution would be an inline const block: `const { .. }`, but it's not yet compatible with our Minimum Supported Rust Version.

Reply:
Alright, changing to just the syntax_set macro.